SYCL/USM/usm_pooling.cpp

```diff
@@ -124,3 +129,13 @@ int main(int argc, char *argv[]) {
 // CHECK-1245-NEXT:  ZE ---> zeMemFree
 // CHECK-1245-NEXT:  ZE ---> [[API]](
 // CHECK-1245-NEXT:  ZE ---> [[API]](
+
+// CHECK-15: Test [[API:zeMemAllocHost|zeMemAllocDevice|zeMemAllocShared]]
+// CHECK-15-NEXT:  ZE ---> [[API]](
+// CHECK-15-NEXT:  ZE ---> zeMemGetAllocProperties
+// CHECK-15-NEXT:  ZE ---> zeMemGetAllocProperties
+// CHECK-15-NEXT:  ZE ---> [[API]](
+// CHECK-15-NEXT:  ZE ---> zeMemGetAllocProperties
+// CHECK-15-NEXT:  ZE ---> zeMemGetAllocProperties
+// CHECK-15-NEXT:  ZE ---> zeMemGetAllocProperties
+// CHECK-15-NEXT:  ZE ---> zeMemFree
```

Comment:
If I get it right, we have 2 zeMemAlloc* calls but only 1 zeMemFree call. Is this expected?

Reply:
Yes, some memory is freed at Context deletion and those frees don't show up in the ZE_DEBUG=1 output. A check with ZE_DEBUG=4 shows no leakage.